examples/herding_basic.py

```diff
@@ -84,57 +80,47 @@ def main(out_path: Path | None = None) -> tuple[float, float, float]:
     coreset_size = 100
 
     # Setup the original data object
-    data = ArrayData.load(x)
+    data = Data(x)
 
     # Set the bandwidth parameter of the kernel using a median heuristic derived from at
     # most 1000 random samples in the data.
     num_samples_length_scale = min(num_data_points, 1_000)
     generator = np.random.default_rng(random_seed)
     idx = generator.choice(num_data_points, num_samples_length_scale, replace=False)
     length_scale = median_heuristic(x[idx])
+    kernel = SquaredExponentialKernel(length_scale=length_scale)
 
     print("Computing herding coreset...")
     # Compute a coreset using kernel herding with a squared exponential kernel.
-    herding_key, sample_key, rp_key, stein_key = random.split(
-        random.key(random_seed), num=4
-    )
-    herding_object = KernelHerding(
-        herding_key, kernel=SquaredExponentialKernel(length_scale=length_scale)
-    )
-    herding_object.fit(
-        original_data=data, strategy=SizeReduce(coreset_size=coreset_size)
-    )
+    sample_key, rpc_key, stein_key = random.split(random.key(random_seed), num=3)
+    herding_solver = KernelHerding(coreset_size, kernel)
+    herding_coreset, _ = eqx.filter_jit(herding_solver.reduce)(data)
```

Comment:
What is the reasoning behind needing ` eqx.filter_jit` wrapping the reduce calls in the examples?

Reply:
If this is necessary should this be included in documentation for the solvers?

Reply:
Demonstrating that you can JIT compile the reduction step. We don't want to automatically apply JIT decoration to the solver reduce methods for a few reasons. Firstly, it leads to a lot of repeat decoration across the codebase. Secondly, and more importantly, we want to defer JIT decoration to the largest possible scope which enables the compiler to perform more/better optimisations.

We use `eqx.filter_jit` instead of `jax.jit` because it provides a more convenient interface (automatically selects jax and numpy arrays as traceable and fixes everything else as static) [see here](https://docs.kidger.site/equinox/api/transformations/#equinox.filter_jit).

Reply:
Makes sense!